pubsub/utils.go

```diff
@@ -88,3 +90,63 @@ func connErr(err error) bool {
 func invalidReplayIDErr(err error) bool {
 	return strings.Contains(strings.ToLower(err.Error()), "replay id validation failed")
 }
+
+func validateAndPreparePayload(dataMap opencdc.StructuredData, avroSchema string) (map[string]interface{}, error) {
+	var schema map[string]interface{}
+	if err := json.Unmarshal([]byte(avroSchema), &schema); err != nil {
+		return nil, err
+	}
+
+	avroRecord := make(map[string]interface{})
+	fields := schema["fields"].([]interface{})
+
+	for _, field := range fields {
+		fieldMap := field.(map[string]interface{})
+		fieldName := fieldMap["name"].(string)
+		fieldType := fieldMap["type"]
```

Comment:
all these castings are not safe, we need to check for `fields, ok := schema["fields"].([]interface{}) if !ok{}`, which could cause a panic